server/app/jobs/container_cleanup_job.rb

```diff
@@ -26,4 +26,28 @@ def destroy_deleted_containers
       c.destroy
     end
   end
+
+  def terminate_ghost_containers
+    Container.unscoped.where(:grid_service_id.ne => nil).each do |c|
+      if c.grid_service.nil?
+        terminate_ghost_container(c)
+      end
+    end
+  end
+
+  # @param [Docker::Container] container
+  def terminate_ghost_container(container)
+    if container.host_node
+      info "terminating #{container.to_path}"
+      terminator = Docker::ServiceTerminator.new(container.host_node)
+      terminator.request_terminate_service(
+        container.grid_service_id.to_s, container.instance_number, {lb: true}
```

Comment:
Thinking about the `{lb: true}`, which will look at the container labels on the agent, and recursively remove the etcd entries for the service if the container is `instance_number=1`....

Could this ghost container be e.g. a leftover container from an earlier service deployment, such that the container's labeled service is still active with other containers, and this would cause their LB configs to be wiped out?